pkg/aws/awscontrolplane/admit_awscontrolplane.go

```diff
@@ -81,11 +83,59 @@ func (a *Admitter) Admit(request *v1beta1.AdmissionRequest) ([]admission.PatchOp
 	}
 
 	var result []admission.PatchOperation
+	var numberOfAZs int
 
-	// We only need to default if attributes are not set
-	if awsControlPlaneCR.Spec.AvailabilityZones != nil && awsControlPlaneCR.Spec.InstanceType != "" {
+	// We only need to manipulate if attributes are not set or it's a create operation
+	if awsControlPlaneCR.Spec.AvailabilityZones != nil && awsControlPlaneCR.Spec.InstanceType != "" && request.Operation != "CREATE" {
 		return result, nil
 	}
+	// We need to fetch the G8sControlPlane in case AZs need to be defaulted or the awscontrolplane is just created
+	if (aws.IsHAVersion(releaseVersion) && awsControlPlaneCR.Spec.AvailabilityZones == nil) || request.Operation == "CREATE" {
+		numberOfAZs = aws.DefaultMasterReplicas
+		fetch := func() error {
+			ctx := context.Background()
+
+			// We try to fetch the G8sControlPlane CR.
+			g8sControlPlane := &infrastructurev1alpha2.G8sControlPlane{}
+			{
+				a.Log("level", "debug", "message", fmt.Sprintf("Fetching G8sControlPlane %s", awsControlPlaneCR.Name))
+				err := a.k8sClient.CtrlClient().Get(ctx,
+					types.NamespacedName{Name: awsControlPlaneCR.GetName(),
+						Namespace: awsControlPlaneCR.GetNamespace()},
+					g8sControlPlane)
```

Comment:
My brain hurts when I see this. What about something more symmetric and simple. With this it is again more easily assessable how many arguments are actually passed. 

```suggestion
				err := a.k8sClient.CtrlClient().Get(
					ctx,
					types.NamespacedName{Name: awsControlPlaneCR.GetName(), Namespace: awsControlPlaneCR.GetNamespace()},
					g8sControlPlane,
				)
```